system_tests/bigquery.py

```diff
@@ -0,0 +1,239 @@
+        dataset.update()
+        self.assertEqual(len(dataset.access_grants), len(after))
+        for found, expected in zip(dataset.access_grants, after):
+            self.assertEqual(found.role, expected.role)
+            self.assertEqual(found.entity_type, expected.entity_type)
+            self.assertEqual(found.entity_id, expected.entity_id)
+
+    def test_list_datasets(self):
+        datasets_to_create = [
+            'new%d' % (1000 * time.time(),),
+            'newer%d' % (1000 * time.time(),),
+            'newest%d' % (1000 * time.time(),),
+        ]
+        for dataset_name in datasets_to_create:
+            dataset = CLIENT.dataset(dataset_name)
+            dataset.create()
+            self.to_delete.append(dataset)
+
+        # Retrieve the datasets.
+        all_datasets, token = CLIENT.list_datasets()
+        self.assertTrue(token is None)
+        created = [dataset for dataset in all_datasets
+                   if dataset.name in datasets_to_create and
+                   dataset.project == CLIENT.project]
+        self.assertEqual(len(created), len(datasets_to_create))
+
+    def test_create_table(self):
+        DATASET_NAME = 'system_tests'
+        dataset = CLIENT.dataset(DATASET_NAME)
+        self.assertFalse(dataset.exists())
+        dataset.create()
+        self.to_delete.append(dataset)
+        TABLE_NAME = 'test_table'
+        full_name = bigquery.SchemaField('full_name', 'STRING',
+                                         mode='REQUIRED')
+        age = bigquery.SchemaField('age', 'INTEGER', mode='REQUIRED')
+        table = dataset.table(TABLE_NAME, schema=[full_name, age])
+        self.assertFalse(table.exists())
+        table.create()
+        self.to_delete.insert(0, table)
+        self.assertTrue(table.exists())
+        self.assertEqual(table.name, TABLE_NAME)
+        self.assertTrue(table._dataset is dataset)
+
+    def test_list_tables(self):
+        DATASET_NAME = 'system_tests'
+        dataset = CLIENT.dataset(DATASET_NAME)
+        self.assertFalse(dataset.exists())
+        dataset.create()
+        self.to_delete.append(dataset)
+        tables_to_create = [
+            'new%d' % (1000 * time.time(),),
+            'newer%d' % (1000 * time.time(),),
+            'newest%d' % (1000 * time.time(),),
+        ]
+        full_name = bigquery.SchemaField('full_name', 'STRING',
+                                         mode='REQUIRED')
+        age = bigquery.SchemaField('age', 'INTEGER', mode='REQUIRED')
+        for table_name in tables_to_create:
+            table = dataset.table(table_name, schema=[full_name, age])
+            table.create()
+            self.to_delete.insert(0, table)
+
+        # Retrieve the tables.
+        all_tables, token = dataset.list_tables()
+        self.assertTrue(token is None)
+        created = [table for table in all_tables
+                   if table.name in tables_to_create and
+                   table._dataset.name == DATASET_NAME]
```

Comment:
RE:

> I added that accessor to make the `job` code cleaner in #1056 (in da68fe3).

Should we file a follow-up issue to make sure it gets changed here as well?